watertap/core/plugins/solvers.py

```diff
@@ -41,27 +40,32 @@ def _pyomo_nl_writer_logger_filter(record):
     "ipopt-watertap",
     doc="The Ipopt NLP solver, with user-based variable and automatic Jacobian constraint scaling",
 )
-class IpoptWaterTAP(IPOPT):
+class IpoptWaterTAP:
+
+    name = "ipopt-watertap"
+    _base_solver = IPOPT
+
     def __init__(self, **kwds):
-        kwds["name"] = "ipopt-watertap"
-        self._cleanup_needed = False
-        super().__init__(**kwds)
-
-    def _presolve(self, *args, **kwds):
-        if len(args) > 1 or len(args) == 0:
-            raise TypeError(
-                f"IpoptWaterTAP.solve takes 1 positional argument but {len(args)} were given"
-            )
-        if not isinstance(args[0], (_BlockData, IBlock)):
-            raise TypeError(
-                "IpoptWaterTAP.solve takes 1 positional argument: a Pyomo ConcreteModel or Block"
-            )
+        kwds["name"] = self.name
+        self.options = Bunch()
+        if kwds.get("options") is not None:
+            for key in kwds["options"]:
+                setattr(self.options, key, kwds["options"][key])
```

Comment:
Incredibly minor suggestion to streamline:

```suggestion
        for opt_key, opt_val in kwds.get("options", {}).items():
            setattr(self.options, opt_key, opt_val)
```